instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpSpanNameExtractor.java

```diff
@@ -42,4 +43,12 @@ public String extract(REQUEST request) {
     }
     return "HTTP request";
   }
+
+  @Nullable
+  private String extractRoute(REQUEST request) {
+    if (attributesExtractor instanceof HttpServerAttributesExtractor) {
+      return ((HttpServerAttributesExtractor<REQUEST, ?>) attributesExtractor).route(request);
+    }
+    return null;
+  }
```

Comment:
is it worth `HttpClientSpanNameExtractor` and `HttpServerSpanNameExtractor`?

Reply:
Hmm, maybe we could have different factory methods in `HttpSpanNameExtractor`? One accepting the client extractor, the other the server one

Reply:
`instanceof` seems OK too from what I can tell